dot/parachain/types/overseer_message.go

```diff
@@ -84,6 +109,26 @@ func (h HypotheticalCandidateComplete) RelayParentHash() common.Hash {
 	return h.CommittedCandidateReceipt.Descriptor.RelayParent
 }
 
+func (h HypotheticalCandidateComplete) GetParentHeadDataHash() (common.Hash, error) {
+	return h.PersistedValidationData.ParentHead.Hash()
```

Comment:
following my previous question, looks like the Incomplete `HypotheticalCandidate` is a middle ground(or status), and those methods are just to implement `HypotheticalCandidate` interface, maybe that's why they are all returning nil?

Reply:
Yeah, in this package we need to deal with two kind of candidates, incompletes and completes, so this interface helps the prospective-parachains to deal with both types.